### ISSUE TITLE
Snowflake: LET variable declarations in scripting blocks not parsable

### ISSUE THREAD

**Issue by humzahussain973** [sqlfluff/sqlfluff#7863]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

SQLFluff fails to parse Snowflake Scripting Language LET variable declarations inside stored procedures, flagging valid SQL as unparsable.

### Expected Behaviour

The following valid Snowflake SQL should lint without parsing errors:
CREATE OR REPLACE PROCEDURE test()
RETURNS VARCHAR
LANGUAGE SQL
AS
BEGIN
  LET x VARCHAR := 'hello';
  RETURN x;
END;

### Observed Behaviour

SQLFluff flags the LET declaration with: Found unparsable section

### How to reproduce

1. Install SQLFluff: pip install sqlfluff
2. Create a test.sql file with the following content:

CREATE OR REPLACE PROCEDURE test()
RETURNS VARCHAR
LANGUAGE SQL
AS
BEGIN
  LET x VARCHAR := 'hello';
  RETURN x;
END;

3. Run: sqlfluff lint test.sql --dialect snowflake
4. Observe parsing error on the LET declaration line

### Dialect

snowflake

### Version

latest

### Configuration

[sqlfluff]
dialect = snowflake

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
With the latest `sqlfluff` I cannot reproduce the parsing error, it parses as `scripting_let_statement`

```bash
sqlfluff parse test.sql -d snowflake
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        create_procedure_statement:
[L:  1, P:  1]      |            keyword:                                          'CREATE'
[L:  1, P:  7]      |            whitespace:                                       ' '
[L:  1, P:  8]      |            keyword:                                          'OR'
[L:  1, P: 10]      |            whitespace:                                       ' '
[L:  1, P: 11]      |            keyword:                                          'REPLACE'
[L:  1, P: 18]      |            whitespace:                                       ' '
[L:  1, P: 19]      |            keyword:                                          'PROCEDURE'
[L:  1, P: 28]      |            whitespace:                                       ' '
[L:  1, P: 29]      |            function_name:
[L:  1, P: 29]      |                function_name_identifier:                     'test'
[L:  1, P: 33]      |            function_parameter_list:
[L:  1, P: 33]      |                bracketed:
[L:  1, P: 33]      |                    start_bracket:                            '('
[L:  1, P: 34]      |                    [META] indent:
[L:  1, P: 34]      |                    [META] dedent:
[L:  1, P: 34]      |                    end_bracket:                              ')'
[L:  1, P: 35]      |            newline:                                          '\n'
[L:  2, P:  1]      |            keyword:                                          'RETURNS'
[L:  2, P:  8]      |            whitespace:                                       ' '
[L:  2, P:  9]      |            data_type:
[L:  2, P:  9]      |                data_type_identifier:                         'VARCHAR'
[L:  2, P: 16]      |            newline:                                          '\n'
[L:  3, P:  1]      |            keyword:                                          'LANGUAGE'
[L:  3, P:  9]      |            whitespace:                                       ' '
[L:  3, P: 10]      |            keyword:                                          'SQL'
[L:  3, P: 13]      |            newline:                                          '\n'
[L:  4, P:  1]      |            keyword:                                          'AS'
[L:  4, P:  3]      |            newline:                                          '\n'
[L:  5, P:  1]      |            scripting_block_statement:
[L:  5, P:  1]      |                keyword:                                      'BEGIN'
[L:  5, P:  6]      |                [META] indent:
[L:  5, P:  6]      |                newline:                                      '\n'
[L:  6, P:  1]      |                statement:
[L:  6, P:  1]      |                    scripting_let_statement:
[L:  6, P:  1]      |                        keyword:                              'LET'
[L:  6, P:  4]      |                        whitespace:                           ' '
[L:  6, P:  5]      |                        variable:                             'x'
[L:  6, P:  6]      |                        whitespace:                           ' '
[L:  6, P:  7]      |                        data_type:
[L:  6, P:  7]      |                            data_type_identifier:             'VARCHAR'
[L:  6, P: 14]      |                        whitespace:                           ' '
[L:  6, P: 15]      |                        assignment_operator:                  ':='
[L:  6, P: 17]      |                        whitespace:                           ' '
[L:  6, P: 18]      |                        expression:
[L:  6, P: 18]      |                            quoted_literal:                   "'hello'"
[L:  6, P: 25]      |                statement_terminator:                         ';'
[L:  6, P: 26]      |                newline:                                      '\n'
[L:  7, P:  1]      |                statement:
[L:  7, P:  1]      |                    return_statement:
[L:  7, P:  1]      |                        keyword:                              'RETURN'
[L:  7, P:  7]      |                        whitespace:                           ' '
[L:  7, P:  8]      |                        expression:
[L:  7, P:  8]      |                            column_reference:
[L:  7, P:  8]      |                                naked_identifier:             'x'
[L:  7, P:  9]      |                statement_terminator:                         ';'
[L:  7, P: 10]      |                newline:                                      '\n'
[L:  8, P:  1]      |                [META] dedent:
[L:  8, P:  1]      |                keyword:                                      'END'
[L:  8, P:  4]      |    statement_terminator:                                     ';'
[L:  8, P:  5]      |    [META] end_of_file:
```

**Comment by peterbud** [comment]
Closing due to lack of response
